### ISSUE TITLE
user.name was not found for .gitconfig with includeIf

### ISSUE THREAD

**Issue by mloskot** [extrawurst/gitui#1696]
I use a [peculiar `.gitconfig` with numerous `includeIf`](https://github.com/mloskot/wsl-config/blob/ad480d54984259a1a28f58bb194b80a9869fc1aa/git/.gitconfig#L1-L14) directives and it works well with `git` and  Fork and GitHub Desktop, as well as with `lazygit`:

```
$ git config --get user.name
Mateusz Łoskot
$ git config --get user.email
mateusz@loskot.net
```

I'm also using `export GPG_TTY=$(tty)` and signing my commits using terminal-based passphrase prompt.

I've just tried `gitui` on Ubuntu 22.04 and it seems tripping over my special Git configuration, complaining with:

```
commit error:
git error:config value 'user.name' was not found; class=Config (7); code=NotFound (-3)
```

I've searched around and I found this about `gitui` not using `git cli`, https://github.com/extrawurst/gitui/issues/74#issuecomment-631522799, so I assume it does not rely on `git`-native processing of `.gitconfig`. This may explain it may be tripping over `includeIf` indeed.

**Comment by pm100** [comment]
gitui does not use the git cli underneath but it does use libgit2 which does support includeif, so it should work. 

is the config file you show the one that failed? Could you please run `gitui -l` to produce a log file. Te log will be here

 - using XDG: $XDG_CACHE_HOME/gitui/gitui.log
 - $HOME/.cache/gitui/gitui.log otherwise

**Comment by mloskot** [comment]
> is the config file you show the one that failed?

Yes, this is my [.gitconfig](https://github.com/mloskot/wsl-config/blob/d3babb60775d476698856b2931e98e59d4a625cf/git/.gitconfig) with `includeIf`-s.

> Could you please run gitui -l to produce a log file

Of course, here is the log file from `$HOME/.cache/gitui/gitui.log` on my Ubuntu 22.04 (on WLS): [gitui.log](https://github.com/extrawurst/gitui/files/11525523/gitui.log)

**Comment by pm100** [comment]
OK, this i complicated issue caused by a bug in libgit2. Your use of %(prefix) is what is causing the error. The temporatry solution is to duplicate the includeifs that you need to use with gitui. I tested this

```
[user]
    name="test"
[includeIf "gitdir:%(prefix)//Users/pm100/**"]
	path = %(prefix)//Users/pm100/mygit
[includeIf "gitdir:/Users/pm100/**"]
	path = /Users/pm100/mygit  
```

and it works fine. libgit should be ignoring the %(prefix) in the non windows (non WSL too) case but it is not (they comment that they should but forget to do it)

**Comment by pm100** [comment]
@mloskot leave it open please, useful for tracking the libgit issue

**Comment by pm100** [comment]
@mloskot I also suspect that running the windows binary under wsl would work, although people have repoerted that this is slow to load in WSL

**Comment by mloskot** [comment]
@pm100 running `gitui.exe` under WSL is failing for me with a new kind of issue

![image](https://github.com/extrawurst/gitui/assets/80741/980e473e-2600-4e25-906d-4c3fe4c9af2a)

A side note regarding the libgit2 bug, interestingly the `%(prefix)`-less workaround from https://github.com/extrawurst/gitui/issues/1696#issuecomment-1556867860 is upsetting lazygit

![image](https://github.com/extrawurst/gitui/assets/80741/138c1b36-f102-4f86-aa82-c61367f9b545)

**Comment by pm100** [comment]
@mloskot windows binary runs for me but behaves oddly :-( , it was worth a shot

**Comment by mloskot** [comment]
@pm100 Yeah, it was. Testing is what I can only offer for now - I can't speak any Rust, yet.
